crates/primitives/convert/src/to_felt.rs

```diff
@@ -92,3 +92,55 @@ impl_for_wrapper!(Nonce);
 impl_for_wrapper!(EntryPointSelector);
 impl_for_wrapper!(CompiledClassHash);
 impl_for_wrapper!(ContractAddressSalt);
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use starknet_core::types::EthAddress;
+
+    #[test]
+    fn test_eth_address_to_felt() {
+        let hex = "0x123456789abcdef0123456789abcdef012345678";
+        let eth_address = EthAddress::from_hex(hex).unwrap();
+        assert_eq!((&eth_address).to_felt(), Felt::from_hex_unchecked(hex));
+        assert_eq!(eth_address.to_felt(), Felt::from_hex_unchecked(hex));
+    }
+
+    #[test]
+    fn test_patricia_key_to_felt() {
+        let key: u128 = 0x123456789abcdef0123456789abcdef;
+        let patricia_key: PatriciaKey = key.into();
+        assert_eq!((&patricia_key).to_felt(), Felt::from(key));
+        assert_eq!(patricia_key.to_felt(), Felt::from(key));
+    }
+
+    #[test]
+    fn test_contract_address_to_felt() {
+        let address: u128 = 0x123456789abcdef0123456789abcdef;
+        let contract_address: ContractAddress = address.into();
+        assert_eq!((&contract_address).to_felt(), Felt::from(address));
+        assert_eq!(contract_address.to_felt(), Felt::from(address));
+    }
+
+    #[test]
+    fn test_h160_to_felt() {
+        let value: [u8; 20] = [
+            0x12, 0x34, 0x56, 0x78, 0x9a, 0xbc, 0xde, 0xf0, 0x12, 0x34, 0x56, 0x78, 0x9a, 0xbc, 0xde, 0xf0, 0x12, 0x34,
+            0x56, 0x78,
+        ];
+        let h160 = H160::from_slice(&value);
+        assert_eq!(h160.to_felt(), Felt::from_bytes_be_slice(&value));
+    }
+
+    #[test]
+    fn test_chain_id_to_felt() {
+        let main_chain_id = ChainId::Mainnet;
+        assert_eq!(main_chain_id.to_felt(), Felt::from_bytes_be_slice(b"SN_MAIN"));
```

Comment:
for the test to be relevant shouldn't we hardcode the Felt value instead of using the `from_bytes_be_slice` method?